user_modules/shared_testcases/testCasesForPerformAudioPassThru.lua

```diff
@@ -0,0 +1,566 @@
+  local CorIdPerfAudioPassThruOnlyMandatoryVD= self.mobileSession:SendRPC("PerformAudioPassThru",
+    {
+      samplingRate = "22KHZ",
+      maxDuration = 500500,
+      bitsPerSample = "16_BIT",
+      audioType = "PCM",
+      audioPassThruIcon =
+      { value = "icon.png",
+        imageType = "STATIC"
+      }
+    })
+
+  -- hmi expects UI.PerformAudioPassThru request
+  EXPECT_HMICALL("UI.PerformAudioPassThru",
+    {
+      appID = self.applications[applicationName],
+      audioPassThruDisplayTexts = {
+        {fieldName = "audioPassThruDisplayText1", fieldText = ""},
+        {fieldName = "audioPassThruDisplayText2", fieldText = ""},
+      },
+      maxDuration = 500500,
+      muteAudio = true,
+      audioPassThruIcon = { imageType = "STATIC", value = "icon.png"}
+
+    })
+  :Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, "UI.PerformAudioPassThru", "SUCCESS", {})
+    end)
+
+  self.mobileSession:ExpectResponse(CorIdPerfAudioPassThruOnlyMandatoryVD, { success = true, resultCode = "SUCCESS",
+    })
+
+end
+
+function testCasesForPerformAudioPassThru:PerformAudioPassThru_Diff_Speech_Capabilities(self, ttsChunksType_value)
+  local ttsChunksType_array = {
+    {text = "4025",type = "PRE_RECORDED"},
+    {text = "Sapi",type = "SAPI_PHONEMES"},
+    {text = "LHplus", type = "LHPLUS_PHONEMES"},
+    {text = "Silence", type = "SILENCE"}
+  }
+  local ttsChunksType = {}
+	
+	for i = 1, #ttsChunksType_array do
+	    -- print (ttsChunksType_array[i].type)
+	    if(ttsChunksType_array[i].type == ttsChunksType_value) then
+	      --print("IN"..i)
+	      ttsChunksType = ttsChunksType_array[i]
+	    end
+	  end
+  
+  --print(ttsChunksType.type, ttsChunksType.text)
+
+  if ( (ttsChunksType.type == "PRE_RECORDED") or
+    (ttsChunksType.type == "SAPI_PHONEMES") or
+    (ttsChunksType.type == "LHPLUS_PHONEMES") or
+    (ttsChunksType.type == "SILENCE")
+  )
+  then
+  --print ("IN")
+    local CorIdPerformAudioPassThruSpeechCap = self.mobileSession:SendRPC("PerformAudioPassThru",
+      {
+        initialPrompt = {
+				          {
+				            text = ttsChunksType.text,
+				            type = ttsChunksType.type
+				          },
+
+        				},
+        audioPassThruDisplayText1 = "DisplayText1",
+        audioPassThruDisplayText2 = "DisplayText2",
+        samplingRate = "8KHZ",
+        maxDuration = 2000,
+        bitsPerSample = "8_BIT",
+        audioType = "PCM",
+        muteAudio = true,
+	        audioPassThruIcon =
+	        { value = "icon.png",
+	          imageType = "STATIC"
+	        }
+
+      })
+
+    --hmi side: expect for TTS.Speak
+    EXPECT_HMICALL("TTS.Speak",
+      {
+        ttsChunks =
+        {
+
+          {
+            text = ttsChunksType.text,
+            type = ttsChunksType.type
+          },
+        }
+
+      })
+    :Do(function(_,data)
+
+        self.hmiConnection:SendResponse(data.id, "TTS.Speak", "UNSUPPORTED_RESOURCE", { })
```

Comment:
@maria-mih, why you simulate HMI response with resulcode = "UNSUPPORTED_RESOURCE"
If the idea is to send "UNSUPPORTED_RESOURCE", you should use 
self.hmiConnection:SendError(data.id, "TTS.Speak", "UNSUPPORTED_RESOURCE", { })